Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift

```diff
@@ -23,6 +23,8 @@ internal class RUMResourceScope: RUMScope, RUMContextProvider {
     private var resourceLoadingStartTime: Date
     /// The HTTP method used to load this Resource.
     private var resourceHTTPMethod: RUMHTTPMethod
+    /// ID of the action which triggered this Resource.
+    private let triggeringActionID: RUMUUID?
```

Comment:
The `activeUserActionID` is part of the `context`. Instead of duplicating this information, we could basically capture the context once at scope initialization time:
```diff
+ let context: RUMContext 
- var context: RUMContext {
-     return parent.context
- }
``` 
as [we fixed this bug in Android SDK](https://github.com/DataDog/dd-sdk-android/commit/b88b5b50062b59d43be437501844ae29ecef42ab#).

By introducing separate `triggeringActionID` value the `self.context` becomes inconsistent which is error-prone.

Reply:
🤔 but this is not duplicate info: the ticket says _"Resources are linked with the wrong Action"_, what you suggest also changes which view is linked with the resource because active view is accessed via current context 😕  
either the ticket is wrong or Android's fix is wrong.?

Reply:
By design, when initializing the `RUMResourceScope` the `context` defines the exact state of the RUM product during Resource start, including active View ID, active Action ID, Session ID and Application ID. None of those should change for the Resource - once we capture the context, it will always describe the right state.

CC @xgouchet ?

Reply:
then ticket is wrong 👌 
i'm changing the ticket first, then the code

Reply:
Sorry if the ticket was misleading. Indeed the goal is to capture what the active context is when the resource is started, so that when it is sent, is has the relevant context information.
A resource is always linked to the context that was active when it started, so that even if another view/action is started before the resource stops, it'll be linked with the original view/action.

Reply:
so `ResourceScope` should not be a `ContextProvider` anymore as it doesn't always have the up-to-date context, correct @ncreated ?

Reply:
> so ResourceScope should not be a ContextProvider anymore as it doesn't always have the up-to-date context

It may still be the `ContextProvider` for architectural consistency. There's nothing invalid about its context (it's never "not up-to-date"). But as `RUMResourceScope` has no children we can remove `ContextProvider` conformance if yo prefer and add it once needed.

Reply:
okay, i removed as it's unused ✅